src/main/java/io/lettuce/core/ReadFrom.java

```diff
@@ -193,6 +195,22 @@ public static ReadFrom valueOf(String name) {
             throw new IllegalArgumentException("Name must not be empty");
         }
 
+        int index = name.indexOf(':');
+
+        if (index != -1) {
+            String type = name.substring(0, index);
+            String value = name.substring(index + 1);
+            if (LettuceStrings.isEmpty(value)) {
+                throw new IllegalArgumentException("Value must not be empty for the type '" + type + "'");
+            }
+            if (type.equalsIgnoreCase("subnet")) {
+                return subnet(value.split(","));
+            }
+            if (type.equalsIgnoreCase("regex")) {
+                return regex(Pattern.compile(value));
```

Comment:
Possibly throws `PatternSyntaxException`

If we want to avoid expanding the contract (this is a runtime exception, but assuming existing applications could be catching the `IllegalArgumentsException` it might constitute a breach of contract to them) we should perhaps wrap the  `PatternSyntaxException` inside a `IllegalArgumentsException`? What do you think?